packages/state-transition/src/block/processConsolidationRequest.ts

```diff
@@ -70,4 +78,56 @@ export function processConsolidationRequest(
     targetIndex,
   });
   state.pendingConsolidations.push(pendingConsolidation);
+
+  // Churn any target excess active balance of target and raise its max
+  if (hasEth1WithdrawalCredential(targetValidator.withdrawalCredentials)) {
+    switchToCompoundingValidator(state, targetIndex);
+  }
+}
+
+/**
+ * Determine if we should set consolidation target validator to compounding credential
+ */
+function isValidSwitchToCompoundRequest(
+  state: CachedBeaconStateElectra,
+  consolidationRequest: electra.ConsolidationRequest
+): boolean {
+  const {sourcePubkey, targetPubkey, sourceAddress} = consolidationRequest;
+  const sourceIndex = state.epochCtx.getValidatorIndex(sourcePubkey);
+  const targetIndex = state.epochCtx.getValidatorIndex(targetPubkey);
+
+  // Verify pubkey exists
+  if (sourceIndex === null) {
+    return false;
+  }
+
+  // Switch to compounding requires source and target be equal
+  if (sourceIndex !== targetIndex) {
+    return false;
+  }
+
+  const sourceValidator = state.validators.getReadonly(sourceIndex);
+  const addressStr = toHex(sourceValidator.withdrawalCredentials.subarray(12));
```

Comment:
same here, does not seem like the hex conversion is required

Reply:
thanks